### PR TITLE
DynamoDB Table Class Cost Optimization query

### PR DESCRIPTION
**Pull request by irmitz**
*Description of changes:*
Created a new folder for "DynamoDB Table Class Cost Optimization"
Added a readme file for the Athena CUR query + the sql file.  

What does the query do? 
This Athena query analyzes the AWS Cost and Usage Report (CUR) data to determine whether switching between the DynamoDB Standard and Standard-IA table classes can be cost-effective for your DynamoDB tables. The query provides insights into the actual storage and throughput costs, as well as recommendations on which tables are good candidates for migrating to the more cost-effective table class.

### CODE REVIEW

File: table_class_optimizer/README.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+## Overview
+DynamoDB's Standard-IA table class lowers storage costs by 60%, but offsets these savings by increasing throughput costs by 25%. This Athena query on [AWS Cost & Usage Reports](https://docs.aws.amazon.com/cur/latest/userguide/what-is-cur.html) (CUR) data help determine whether switching between Standard and Standard-IA table classes is cost-effective for your DynamoDB tables based on your usage and parameters provided. For more information about table classes and their commonalities, see our [best practice guide](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/CostOptimization_TableClass.html) in the developer documentation.
+
+To get a summary of your total potential savings at the payer level, use the query with **SUMMARY** set in report_type parameter
+To get a detailed cost optimization report on a per-table level, use the query with **DETAILED** set in report_type parameter
+
+Use the full user guide to effectively use the query.
+
+## User guide
+### Execution steps
```
Comment (switch180) [inline table_class_optimizer/README.md:25]: Add a new section on how to enable CUR. Be sure to explain the options they should set. Or, refer to documentation and then give the parameters (e.g. granularity required) for the user to input.

**Comment by switch180** [comment]
Hi! Here are my requested changes:
- Can you please review the Standard-IA calculations using the pricing in APN1 and maybe APN2? I can't recall, but I have heard that some regions don't have the same breakeven percent due to the disparity between the cost for storage and capacity units. 
- review each code comment I added and address feedback

**Review by switch180 — CHANGES_REQUESTED** [review]

File: table_class_optimizer/README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+**What are the feature highlights?**
+- New table storage class beneficial for those existing DynamoDB workloads where access to the data is very infrequent, but storage size is quite large. Particularly if storage is the dominant cost driver for a table, Standard_IA can help:
+  - Reduce Storage Costs by 60%
+  - With Throughput Cost increasing by 25%
+  - Same cost ratios apply irrespective of Provisioned/On-Demand Capacity Modes, Global Tables or Single region tables
+
+**When is it recommended to switch back to Standard?**
+- When you switch from Standard-IA to Standard, you will pay 20% less on the throughput (read/write) cost and 150% more on the storage cost. Particularly if throughput is the clear dominant cost driver for a table. The rule of thumb is if your throughput cost is x7.5 than the storage cost, Standard will be a be a more cost-effective table class for your usage.
+
+**How can this help AWS customers running large workloads on DynamoDB?**
```
[the remark this answers is not in the document]
Comment (irmitz) [reply table_class_optimizer/README.md:80]: This section is the heart of determining if to switch back to Standard (The opposite decision). If you built your table on S-IA (or switched from Std), you will be paying 150% more on storage (compares of what you are paying while you are on S-IA) and 20% less on throughput if you switch to Std. In some situations when the workload changed (mainly after massive deletion), switching back to Std may be more cost-effective. The ratio is that your storage cost should be less than 13% of your throughput cost in order to benefit from the switch (I presented the full math behind that on the TFC Monthly call in May, if you recall).

File: table_class_optimizer/README.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+## The query
+Download the [DDB_TableClassReco.sql](DDB_TableClassReco.sql) file.
+
+## FAQ
+**What are the feature highlights?**
+- New table storage class beneficial for those existing DynamoDB workloads where access to the data is very infrequent, but storage size is quite large. Particularly if storage is the dominant cost driver for a table, Standard_IA can help:
+  - Reduce Storage Costs by 60%
+  - With Throughput Cost increasing by 25%
+  - Same cost ratios apply irrespective of Provisioned/On-Demand Capacity Modes, Global Tables or Single region tables
```
[the remark this answers is not in the document]
Comment (irmitz) [reply table_class_optimizer/README.md:75]: 1) The query is based on averages on all regions (~25% increase on throughput and exactly 60% discount on storage for all regions), which is the average of all regions. The goal is not to provide $1 level accuracy; it should identify major cost saving opportunities. This is why I set the default parameter "min_savings_per_month"= 50 to eliminate "noises" due to region pricing differences. 

2) The query ignores recommendation (technically set the recommendation—"Optimized") if the reservation is in use.

File: table_class_optimizer/README.md (hunk 1 above)
Comment (irmitz) [reply table_class_optimizer/README.md:25]: Will add CUR setup instructions and references

File: table_class_optimizer/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (switch180) [reply table_class_optimizer/README.md:80]: My comment is on line 80, a different FAQ item

**Review by switch180 — APPROVED** [review]

**Comment by switch180** [comment]
As part of this release, I need to deprecate the old Python tool. 
I have made changes to the code, restructuring the Readme, but still need to make changes to the rest of the package for this release.

**Pull request merged**